<?xml version="1.0"?>

<launch>

    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find robot_control)/world/harbor.world"/>
      <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="gui" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="debug" value="false"/>
    </include>

    
    <group ns="larger_robot">
        <include file="$(find robot_control)/launch/larger_robot.launch" ></include>
        <node name="Localization" pkg="robot_control" type="localization" output="screen" args="larger_robot" required="true"/>
        <node name="ImageProcessing" pkg="robot_control" type="image_processing" output="screen" args="larger_robot" required="true"/>
        <node name="Laser" pkg="robot_control" type="laser" output="screen" args="larger_robot" required="true"/>
        <node name="Mapping" pkg="robot_control" type="mapping" output="screen" args="larger_robot" required="true"/>
        <node name="Navigation" pkg="robot_control" type="navigation" output="screen" args="larger_robot" required="true"/>
        <node name="Control" pkg="robot_control" type="control" output="screen" args="larger_robot" required="true"/>
    </group>
  <!--
    launch-prefix="valgrind" 
    -->
    <group ns="smaller_robot">
        <include file="$(find robot_control)/launch/smaller_robot.launch" ></include>
        <node name="Localization" pkg="robot_control" type="localization" output="screen" required="true" args="smaller_robot"/>
        <node name="ImageProcessing" pkg="robot_control" type="image_processing" output="screen" args="smaller_robot" required="true"/>
        <node name="Laser" pkg="robot_control" type="laser" output="screen" required="true" args="smaller_robot"/>
        <node name="Mapping" pkg="robot_control" type="mapping" output="screen" required="true" args="smaller_robot"/>
        <node name="Navigation" pkg="robot_control" type="navigation" output="screen" required="true" args="smaller_robot"/>
        <node name="Control" pkg="robot_control" type="control" output="screen" required="true" args="smaller_robot"/>
    </group>
    <!--
    -->
    <node name="Application" pkg="robot_control" type="application" output="screen" required="true"/>
    <node name="Knowledge" pkg="robot_control" type="knowledge" output="screen" required="true"/>
    <node name="PathPlanning" pkg="robot_control" type="path_planning" output="screen" required="true"/>

</launch>